tests/frontend/helper.js

```diff
@@ -197,6 +217,38 @@ var helper = {};
     return deferred;
   }
 
+  /**
+   * Same as `waitFor` but using Promises
+   *
+   */
+  helper.waitForPromise = function(conditionFunc, _timeoutTime, _intervalTime){
```

Comment:
`helper.waitFor` already returns a "thenable" (a [Deferred](https://api.jquery.com/category/deferred-object/) object), so callers can already do `await helper.waitFor(...)`, `helper.waitFor(...).then(...)`, etc.

If you need something that doesn't allow the caller to control the Deferred object's state, you can simplify this function:

```javascript
  helper.waitForPromise = function(conditionFunc, _timeoutTime, _intervalTime) {
    return this.waitFor(conditionFunc, _timeoutTime, _intervalTime)).promise();
  }
```

The above returns a jQuery Promise object. If you need a standard Promise object for some reason, or if you want to improve readability, you can do this:

```javascript
  helper.waitForPromise = async function(conditionFunc, _timeoutTime, _intervalTime) {
    return await this.waitFor(conditionFunc, _timeoutTime, _intervalTime);
  }
```

Reply:
Thanks for the input! Trying your last two options the tests in 
https://github.com/ether/etherpad-lite/pull/4408/files#diff-4c39b0646494534de0cbe94479400dabR190
https://github.com/ether/etherpad-lite/pull/4408/files#diff-4c39b0646494534de0cbe94479400dabR201
https://github.com/ether/etherpad-lite/pull/4408/files#diff-4c39b0646494534de0cbe94479400dabR226
fail.

I will try again later, but in case you see a problem in the helper.js tests, it would definitely help :) I don't want to have a massive amount of try-catch

Reply:
Hmm, interesting. I'll take a look.

Reply:
OK, I figured it out: `helper.waitFor()` will throw an uncatchable exception unless you register a callback via the `.fail()` method. Please see the new commits I pushed.